core/src/main/java/org/apache/iceberg/CatalogProperties.java

```diff
@@ -119,6 +119,8 @@ private CatalogProperties() {}
       "client.pool.cache.eviction-interval-ms";
   public static final long CLIENT_POOL_CACHE_EVICTION_INTERVAL_MS_DEFAULT =
       TimeUnit.MINUTES.toMillis(5);
+  /** Name of the custom {@link ClientPool} implementation class. */
+  public static final String CLIENT_POOL_IMPL = "client.pool.impl";
```

Comment:
I actually like dot better, but maybe we should make it with a hyphen like the other impl properties?